ably-ios/ARTRest.m

```diff
@@ -92,6 +95,10 @@ - (instancetype)initWithKey:(NSString *) key {
     return [self initWithOptions:[[ARTClientOptions alloc] initWithKey:key]];
 }
 
+- (void)dealloc {
+    [self.logger debug:__FILE__ line:__LINE__ message:@"%p dealloc", self];
```

Comment:
Yes, but when called, because you're overriding the parent's implementation, which actually deallocates the object, if you don't pass the method up the inheritance chain, the deallocation won't happen and there'll be a memory leak.

See https://developer.apple.com/library/ios/documentation/Cocoa/Conceptual/MemoryMgmt/Articles/mmPractical.html, the third example under `Don’t Use Accessor Methods in Initializer Methods and dealloc`.

At least this is what I understand from that, I've never worked with dealloc.